lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2304,6 +2298,16 @@ protected static function compute_style_properties( $styles, $settings = array()
 				continue;
 			}
 
+			// Look up protected properties, keyed by value path.
+			// Skip protected properties that are explicitly set to `null`.
+			$path_string = implode( '.', $value_path );
```

Comment:
Do we need to retain the `is_array()` check here?

Reply:
This is moved to line 2268.

Reply:
Oh, I see. I don't know enough about this function, but is it possible that `$value_path` is sometimes a string instead of an array? If so, previously those would have still been processed where now they would not because of the early `continue` at line :2268.

Reply:
> is it possible that $value_path is sometimes a string instead of an array?

I don't believe so. At least, `compute_style_properties` is never passed any value for `$properties` inside this class. Also, my testing shows that if `get_property_value` were passed a string instead of an array it would silently fail with no global styles being output, which would happen if $value_path were a string.

Reply:
These line also just guesses that value_path is an array. 

```
if ( 'background' === $value_path[0] && isset( $styles['background'] ) ) {
```

If for some reason `$value_path` was a string, it would get the first character of that string. This seems weird and wrong to me.

I think ensuring this code is a type safe is a good thing, but if we want to be safe, then I can refactor and moving the logic down.

Reply:
With this question resolved, I think this is good to commit.